test-apps/imodel-transformer/src/IModelHubUtils.ts

```diff
@@ -62,38 +62,58 @@ export namespace IModelHubUtils {
+  /** Call the specified function for each changeSet of the specified iModel. */
+  export async function forEachChangeSet(requestContext: AuthorizedClientRequestContext, iModelId: GuidString, func: (c: ChangeSet) => void): Promise<void> {
     const changeSets = await IModelHost.iModelClient.changeSets.get(requestContext, iModelId);
-    Logger.logInfo(loggerCategory, `changeSets.length=${changeSets.length}`);
-    if (changeSets.length > 0) {
-      for (const changeSet of changeSets) {
-        Logger.logInfo(loggerCategory, `id="${changeSet.id}", description="${changeSet.description}", fileSize=${changeSet.fileSizeNumber}`);
-      }
+    for (const changeSet of changeSets) {
+      func(changeSet);
     }
   }
 
-  export async function logNamedVersions(requestContext: AuthorizedClientRequestContext, iModelId: GuidString, loggerCategory: string): Promise<void> {
+  /** Call the specified function for each (named) Version of the specified iModel. */
+  export async function forEachNamedVersion(requestContext: AuthorizedClientRequestContext, iModelId: GuidString, func: (v: Version) => void): Promise<void> {
     const namedVersions = await IModelHost.iModelClient.versions.get(requestContext, iModelId);
-    Logger.logInfo(loggerCategory, `namedVersions.length=${namedVersions.length}`);
-    if (namedVersions.length > 0) {
-      for (const namedVersion of namedVersions) {
-        Logger.logInfo(loggerCategory, `id="${namedVersion.id}", changeSetId="${namedVersion.changeSetId}", name="${namedVersion.name}"`);
-      }
+    for (const namedVersion of namedVersions) {
+      func(namedVersion);
     }
   }
 
-  export async function downloadAndOpenBriefcase(requestContext: AuthorizedClientRequestContext, sourceContextId: GuidString, sourceIModelId: GuidString, asOfVersion: IModelVersion): Promise<BriefcaseDb> {
-    const briefcaseProps = await BriefcaseManager.downloadBriefcase(requestContext, {
-      contextId: sourceContextId,
-      iModelId: sourceIModelId,
-      asOf: asOfVersion.toJSON(),
-    });
+  export async function downloadAndOpenBriefcase(requestContext: AuthorizedClientRequestContext, briefcaseArg: RequestNewBriefcaseArg): Promise<BriefcaseDb> {
+    let briefcaseQuery = new BriefcaseQuery().ownedByMe();
+    if (briefcaseArg.briefcaseId) {
+      briefcaseQuery = briefcaseQuery.filter(`BriefcaseId+eq+${briefcaseArg.briefcaseId}`);
+    }
+    const briefcases = await IModelHost.iModelClient.briefcases.get(requestContext, briefcaseArg.iModelId, briefcaseQuery);
+    if (0 === briefcases.length) {
+      const briefcaseProps = await BriefcaseManager.downloadBriefcase(requestContext, briefcaseArg);
+      return BriefcaseDb.open(requestContext, {
+        fileName: briefcaseProps.fileName,
+        readonly: briefcaseArg.briefcaseId ? briefcaseArg.briefcaseId === BriefcaseIdValue.Unassigned : undefined,

```

Comment:
probably clearer to use `false` rather than `undefined`

Reply:
done